platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/files/DataFileMetadata.java

```diff
@@ -132,18 +133,15 @@ public DataFileMetadata(Path file) throws IOException {
             }
         }
 
-        // Initialize this object
-        this.index = index;
-        this.creationDate = Instant.ofEpochSecond(creationSeconds, creationNanos);
-        this.itemsCount = itemsCount;
-        this.compactionLevel = compactionLevel;
+        return new DataFileMetadata(
+                itemsCount, index, Instant.ofEpochSecond(creationSeconds, creationNanos), compactionLevel);
     }
 
-    void writeTo(final BufferedData out) {
-        ProtoWriterTools.writeDelimited(out, FIELD_DATAFILE_METADATA, fieldsSizeInBytes(), this::writeFields);
+    int getFieldsSizeInBytes() {
+        return fieldsSizeInBytes;
     }
 
-    private void writeFields(final WritableSequentialData out) {
+    void writeFields(final WritableSequentialData out) {
```

Comment:
Please, add a JavaDoc that explains what this method writes (only fields, but not a tag / size)

Reply:
I would probably keep `writeTo()` as the method name. `getFieldsSizeInBytes()` can be renamed to `getSizeInBytes()`. It would be consistent with other classes in our code, e.g. `VirtualLeafBytes`